src/appleseed/renderer/kernel/rendering/baserenderer.cpp

```diff
@@ -0,0 +1,225 @@
+
+//
+// This source file is part of appleseed.
+// Visit http://appleseedhq.net/ for additional information and resources.
+//
+// This software is released under the MIT license.
+//
+// Copyright (c) 2015 Esteban Tovagliari, The appleseedhq Organization
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+// Interface header.
+#include "baserenderer.h"
+
+// appleseed.renderer headers.
+#include "renderer/global/globallogger.h"
+#include "renderer/modeling/project/project.h"
+#include "renderer/modeling/scene/scene.h"
+#ifdef APPLESEED_WITH_OIIO
+#include "renderer/kernel/rendering/oiioerrorhandler.h"
+#endif
+#ifdef APPLESEED_WITH_OSL
+#include "renderer/kernel/rendering/rendererservices.h"
+#include "renderer/kernel/shading/closures.h"
+#endif
+
+// appleseed.foundation headers.
+#include "foundation/platform/compiler.h"
+
+// Standard headers.
+#include <cassert>
+#include <exception>
+#include <string>
+
+using namespace foundation;
+using namespace std;
+
+namespace renderer
+{
+
+//
+// BaseRenderer class implementation.
+//
+
+BaseRenderer::BaseRenderer(
+    Project&                project,
+    const ParamArray&       params)
+  : m_project(project)
+  , m_params(params)
+{
+#ifdef APPLESEED_WITH_OIIO
+
+    RENDERER_LOG_DEBUG("%s.", "creating OpenImageIO error handler");
```

Comment:
Although, it's in line with the other messages that announce the creation of the various OSL/OIIO components... Why not.